appinventor/appengine/src/com/google/appinventor/server/project/youngandroid/YoungAndroidProjectService.java

```diff
@@ -842,8 +842,8 @@ private String getCurrentHost() {
         return appengineHost.get();
       }
     } else {
-      // TODO(user): Figure out how to make this more generic
-      return "localhost:8888";
+      // No more hard-coded port number for DevServer; read it from appengine-web.xml instead.
+      return appengineHost.get();
```

Comment:
Rather than doing this, we should get the `appengineHost` and see if it is empty or not. If it is not empty, return the value. If it is empty, check the production flag and return the computed value on line 840 if in production or `"localhost:8888"` if development.